internal/pkg/telemetry/ddtracer.go

```diff
@@ -39,11 +41,15 @@ type stackTracer interface {
 func (t *wrappedDDTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
 	ctx, span := t.tracer.Start(ctx, spanName, opts...)
 	if span != nil {
-		span = &wrappedDDSpan{Span: span}
+		span = &wrappedDDSpan{Span: span, tracerProvider: t.tracerProvider}
 	}
 	return ctx, span
 }
 
+func (s *wrappedDDSpan) TracerProvider() trace.TracerProvider {
+	return s.tracerProvider
+}
+
```

Comment:
Ako sme [riesili tu](https://github.com/keboola/keboola-as-code/pull/1331/commits/098856ecce9d8ba78b56a494cafec440e18d247d), ze datadog OTEL telemetria je trocha zabugovana, a pri zavolani `TracerProvider.Tracer()` sa spusta znova - resetuje.

Tak som narazil este na jednu nevyriesenu vec:
- `TracerProvider` sa da ziskat zo `Span.TracerProvider()`.
- Tato metoda nebola prepisana, a tak vratila povodvny DD tracerProvider, ktory ma uvedeny ^^^ problem.
- :red_circle:  pred zmenou
- :green_circle: po zmene

![image](https://github.com/keboola/keboola-as-code/assets/19371734/28fb617f-323a-4bc6-9100-3baaa002bbcd)

Ostatne zmeny su o tom, a by sa do `wrappedDDSpan` dostal `TracerProvider`, ktory chceme vratit.

Reply:
Prejavovalo sa to tak, ze pri kazdej operacii v `go-cloud` libe sa zalogovalo `DATADOG CONFIGURATION: ...`, tj. cely tracer sa inicializoval znova.